src/shogun/modelselection/GradientModelSelection.cpp

```diff
@@ -218,6 +224,8 @@ CParameterCombination* CGradientModelSelection::select_model(bool print_state)
 		// optimization algorithm
 		nlopt_opt opt=nlopt_create(NLOPT_LD_MMA, total_variables);
 
+		// currently we assume all parameters are positive
+		// (this is NOT true when inducing points and Full Matrix GaussianARDKernel are optimized)
```

Comment:
This will be addressed in the follow-up PRs which will take account of `parameter constraints` and `parameter transformations`.